src/datatypes/datetime.coffee

```diff
@@ -23,6 +23,7 @@ module.exports.DateTime = class DateTime
       null
 
   @fromDate: (date, timezoneOffset) ->
+    if (date instanceof DateTime) then return date
```

Comment:
On line 89 below there is the function:
```
convertToTimezoneOffset: (timezoneOffset = 0) ->
  d = DateTime.fromDate(@toJSDate(), timezoneOffset)
  d.reducedPrecision(@getPrecision())
```

With this new logic, that will no longer alter the timezoneOffset of a provided `DateTime`.  Should this instead first set the timezoneOffset before returning `date`?

Reply:
I believe `@toJSDate()` first converts the DateTime to a javascript `Date` object before it gets passed into the `fromDate` function. So the new logic (I believe) will not alter the functionality of `convertToTimezoneOffset`.